### ISSUE TITLE
Pigment Css not working with Next js basic setup, causing error for 'babelMerge'

### ISSUE THREAD

**Issue by nunesunil** [mui/pigment-css#257]
### Steps to reproduce

clone https://github.com/nunesunil/mui-pigmentCss

### Current behavior

gives the error - 

`TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received an instance of Object
    at Object.isAbsolute (node:path:1188:5)
    at standardizeName (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@babel+core@7.25.7/node_modules/@babel/core/lib/config/files/plugins.js:76:15)
    at resolveAlternativesHelper (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@babel+core@7.25.7/node_modules/@babel/core/lib/config/files/plugins.js:81:28)
    at resolveAlternativesHelper.next (<anonymous>)
    at resolveStandardizedNameForRequire (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@babel+core@7.25.7/node_modules/@babel/core/lib/config/files/plugins.js:159:16)
    at resolveStandardizedName (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@babel+core@7.25.7/node_modules/@babel/core/lib/config/files/plugins.js:182:12)
    at /Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/babel-merge@3.0.0_@babel+core@7.25.7/node_modules/babel-merge/src/index.js:14:24
    at Array.find (<anonymous>)
    at /Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/babel-merge@3.0.0_@babel+core@7.25.7/node_modules/babel-merge/src/index.js:13:28
    at Array.reduce (<anonymous>)
    at mergeArray (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/babel-merge@3.0.0_@babel+core@7.25.7/node_modules/babel-merge/src/index.js:10:36)
    at babelMerge (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/babel-merge@3.0.0_@babel+core@7.25.7/node_modules/babel-merge/src/index.js:34:19)
    at merge (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/options/buildOptions.js:22:42)
    at Array.reduce (<anonymous>)
    at buildOptions (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/options/buildOptions.js:33:95)
    at buildConfigs (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/transform/Entrypoint.helpers.js:68:52)
    at Object.loadAndParse [as loadAndParseFn] (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/transform/Entrypoint.helpers.js:128:7)
    at new Entrypoint (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/transform/Entrypoint.js:40:112)
    at Entrypoint.innerCreate (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/transform/Entrypoint.js:129:27)
    at /Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/transform/Entrypoint.js:85:47
    at EventEmitter.perf (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/utils/EventEmitter.js:45:20)
    at Entrypoint.create (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/transform/Entrypoint.js:84:25)
    at Entrypoint.createRoot (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/transform/Entrypoint.js:62:32)
    at Object.transform (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@wyw-in-js+transform@0.5.4_typescript@5.6.2/node_modules/@wyw-in-js/transform/lib/transform.js:98:45)
    at Object.transform (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/@pigment-css+unplugin@0.0.24_@types+react@18.3.11_react@18.3.1_typescript@5.6.2/node_modules/@pigment-css/unplugin/build/index.js:268:40)
    at Object.transform (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/unplugin@1.14.1/node_modules/unplugin/dist/webpack/loaders/transform.js:120:38)
    at LOADER_EXECUTION (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/next@14.2.14_@babel+core@7.25.7_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4134)
    at runSyncOrAsync (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/next@14.2.14_@babel+core@7.25.7_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4145)
    at iterateNormalLoaders (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/next@14.2.14_@babel+core@7.25.7_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5782)
    at iterateNormalLoaders (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/next@14.2.14_@babel+core@7.25.7_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5650)
    at /Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/next@14.2.14_@babel+core@7.25.7_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:5879
    at r.callback (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/next@14.2.14_@babel+core@7.25.7_react-dom@18.3.1_react@18.3.1__react@18.3.1/node_modules/next/dist/compiled/loader-runner/LoaderRunner.js:1:4039)
    at Object.transform (/Users/sunil/Desktop/mui-pigmentCss/node_modules/.pnpm/unplugin@1.14.1/node_modules/unplugin/dist/webpack/loaders/transform.js:135:5)`

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>
</details>

```
  System:
    OS: macOS 15.0
  Binaries:
    Node: 20.17.0 - /usr/local/bin/node
    npm: 10.8.2 - /usr/local/bin/npm
    pnpm: 9.12.0 - ~/Library/pnpm/pnpm
  Browsers:
    Chrome: 129.0.6668.71
    Edge: Not Found
    Safari: 18.0
  npmPackages:
    @mui/material: ^6 => 6.1.2 
    @mui/material-pigment-css: ^6 => 6.1.2 
    @pigment-css/nextjs-plugin: ^0 => 0.0.24 
    @types/react: ^18 => 18.3.11 
    react: ^18 => 18.3.1 
    react-dom: ^18 => 18.3.1 
    typescript: ^5 => 5.6.2 
```

**Search keywords**: dependencies

**Comment by kcvdk3101** [comment]
I got the same error!

**Comment by kcvdk3101** [comment]
Pigment CSS is not working with Vite + React + TypesScript too

**Comment by santi-m** [comment]
For a quick fix. you can pin `@babel/core` to `7.25.2` with `npm install --save-dev @babel/core@7.25.2`
I think there is something with the new release of babel core
https://github.com/babel/babel/releases/tag/v7.25.7

**Comment by mnajdova** [comment]
Correct, it's related to the latest @babel/core release, we have a similar issue here: https://github.com/mui/material-ui/issues/43971#issuecomment-2391431726. Please pin @babel/core, until we find a fix for it.

**Comment by nicolasmingo** [comment]
Hi, 

I fixed the problem by following this post (https://stackoverflow.com/questions/56098779/how-to-fix-module-build-failed-from-node-modules-babel-loader-lib-index-js)

```shell
yarn add --dev @babel/preset-env
yarn add --dev @babel/core 
```

which updates my **package.json** like this:
```json
"devDependencies": {
    "@babel/core": "^7.25.8",
    "@babel/preset-env": "^7.25.8",
}
```

And ensure you have the **.babelrc** file like this:    
```json
{
    "presets": [
        "@babel/preset-env"
    ]
}
```

**Comment by mnajdova** [comment]
The issue has been resolved in the [v7.25.8](https://github.com/babel/babel/releases/tag/v7.25.8) of @babel/core, as the team decided to backport the old API for resolvePreset and resolvePreset in https://github.com/babel/babel/pull/16888. Long term, the code in wyw could be update to adhear to the new API. cc @brijeshb42
